pkg/application/mqtt_consumer.go

```diff
@@ -65,3 +47,190 @@ func VerifyError(err error) {
+//			"raw":    4,
+//			"int64":  5,
+//			"uint":   6,
+//			"double": 7,
+//		}
+//
+//		err := json.Unmarshal([]byte(msg.Payload()), &capturedData)
+//		if err != nil {
+//			fmt.Println("Erro ao converter JSON:", err)
+//			return
+//		}
+//
+//		// Encontrar o tipo de dispositivo com base no sensorId
+//		var deviceType string
+//		for _, config := range deviceConfiguration {
+//			for _, device := range config.Config {
+//				sensorType := device.Schema.ValueType
+//				for _, dataMap := range capturedData.Data {
+//					if device.SensorID == int(dataMap["sensorId"].(float64)) {
+//						for key, value := range hexMap {
+//							if sensorType == value {
+//								deviceType = key
+//							}
+//						}
+//					}
+//				}
+//			}
+//		}
+//
+//		// Construir a estrutura final removendo os campos indesejados
+//		var finalData entities.CapturedData
+//		dataId := capturedData.ID
+//		for _, dataMap := range capturedData.Data {
+//			var dataRow entities.Row
+//
+//			sensorId, _ := dataMap["sensorId"].(float64)
+//			finalData.ID = int(sensorId)
+//
+//			switch deviceType {
+//
+//			case "int":
+//				value, _ := dataMap["value"].(float64)
+//				dataRow.Value = int(value)
+//			case "float":
+//				value, _ := dataMap["value"].(float64)
+//				dataRow.Value = value
+//			case "bool":
+//				value, _ := dataMap["value"].(bool)
+//				dataRow.Value = value
+//			case "raw":
+//				value, _ := dataMap["value"].(string)
+//				dataRow.Value = value
+//			case "int64":
+//				value, _ := dataMap["value"].(float64)
+//				dataRow.Value = int64(value)
+//			case "uint":
+//				value, _ := dataMap["value"].(float64)
+//				dataRow.Value = uint(value)
+//			case "double":
+//				value, _ := dataMap["value"].(float64)
+//				dataRow.Value = value
+//			default:
+//				fmt.Println("Erro no id")
+//			}
+//
+//			timestamp, _ := dataMap["timestamp"].(string)
+//			dataRow.Timestamp = timestamp
+//
+//			// Outros campos podem ser tratados da mesma maneira
+//
+//			finalData.Rows = append(finalData.Rows, dataRow)
+//		}
+//
+//		// Imprimir os dados decodificados
+//		fmt.Println("CapturedData:", dataId)
+//		for _, row := range finalData.Rows {
+//			fmt.Println("SensorId:", finalData.ID)
+//			fmt.Println("Value:", row.Value)
+//			fmt.Println("Timestamp:", row.Timestamp)
+//		}
+//
+//		transmissionChannel <- finalData
+//	}
+func onMessageReceived(msg mqtt.Message, transmissionChannel chan entities.CapturedData, deviceConfiguration map[string]entities.Device, mqttDeviceConfiguration entities.DeviceConfig) {
+	var data map[string]interface{}
+	var capturedData struct {
+		ID   int                      `json:"id"`
+		Data []map[string]interface{} `json:"data"`
+	}
+
+	err := json.Unmarshal([]byte(msg.Payload()), &data)
+	fmt.Println(data)
+	if err != nil {
+		fmt.Println("Erro ao converter JSON:", err)
+		return
+	}
+
+	for _, config := range mqttDeviceConfiguration.Config {
+		var finalData entities.CapturedData
+		value, _ := data[config.Value]
```

Comment:
Verificar se `value`, `name`, `times` e `sensorType` estão presentes no `map`.
```suggestion
		value, ok := data[config.Value]
		if !ok {
			continue
		}
```